@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -486,14 +470,29 @@ export class PoiRenderer {
      * @param env - TODO! The current zoomLevel level of {@link MapView}
      *
      * @returns `True` if the space is not already allocated by another object (text label or POI)
+     * or a Promise if the image has to be loaded first. This happens when the user explicitly
+     * doesn't load the image when it is put into the cache.
      */
-    prepareRender(pointLabel: TextElement, env: Env): boolean {
+    prepareRender(pointLabel: TextElement, env: Env): boolean | Promise<boolean> {
         const poiInfo = pointLabel.poiInfo;
         if (poiInfo === undefined) {
             return false;
         }
+        const { imageItem, imageTexture } = this.imageCached(poiInfo.imageTextureName);
```

Comment:
There's no need to do this if the poiInfo.buffer is already created (see the if below). Why did you move this out of preparePoi?

Reply:
I wanted to centralise the point where the poiInfo.isValid = false is set, it is just set in this place, as opposed to being set in two places previously.

And it simplifies the `preparePoi` a bit.

But good point about the fact that I don't need this if the buffer is already created, I'll think about this.

Reply:
I moved it, so will resolve.